make-file-list-msys2.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/sh
+
+die () {
+	echo "$*" >&2
+	exit 1
+}
+
+test -n "$ARCH" &&
+test -n "$BITNESS" ||
+die "Need ARCH and BITNESS to be set"
+
+pacman_list () {
+	package_list=$(for arg
+		do
+			pactree -u "$arg"
+		done |
+		grep -v "^\\($(echo $PACKAGE_EXCLUDES | sed \
+			-e 's/ /\\|/g' \
+			-e 's/mingw-w64-/&\\(i686\\|x86_64\\)-/g')\\)\$" |
+		sort |
+		uniq) &&
+	if test -n "$PACKAGE_VERSIONS_FILE"
+	then
+		pacman -Q $package_list >"$PACKAGE_VERSIONS_FILE"
+	fi &&
+	pacman -Ql $package_list |
+	grep -v '/$' |
+	sed 's/^[^ ]* //'
+}
+
+install_required () {
+	# TODO some of these might not be wanted
+	# Packages that have been added after Git SDK 1.0.0 was released...
+	required=
+	for req in mingw-w64-$ARCH-git-credential-manager $SH_FOR_REBASE
+	do
+		test -d /var/lib/pacman/local/$req-[0-9]* ||
+		test -d /var/lib/pacman/local/$req-git-[0-9]* ||
+		required="$required $req"
+	done
+	test -z "$required" ||
+	pacman -Sy --noconfirm $required >&2 ||
+	die "Could not install required packages: $required"
+}
+
+this_script_dir="$(cd "$(dirname "$0")" && pwd -W)" ||
+die "Could not determine this script's dir"
+
+install_required
+
+SH_FOR_REBASE=dash
+
+BASE_PACKAGES="$(pacman -Qg base | awk '{print $2}' | tr '\n' ' ')"
+GIT_PACKAGES="mingw-w64-$ARCH-git mingw-w64-$ARCH-git-credential-manager git-extra openssh"
+UTIL_PACKAGES=
+packages="$BASE_PACKAGES $GIT_PACKAGES $UTIL_PACKAGES"
+
+# TODO need this? should be a parameter? Use BITNESS? 
+PACKAGE_EXCLUDES="db info heimdal git util-linux curl git-for-windows-keyring
+	mingw-w64-p11-kit filesystem msys2-launcher-git rebase"
+# TODO 05-home-dir.post required by filesystem package but is missing from SDK
+# TODO should we exlude all the tz stuff?
```

Comment:
Lots of stuff like perl (needed or git) Local and TZ files really bloat the zip but I don;t kno when might be needed